internal/sequencerapi/api.go

```diff
@@ -0,0 +1,96 @@
+package sequencerapi
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/internal/ethapi"
+	"github.com/ethereum/go-ethereum/metrics"
+	"github.com/ethereum/go-ethereum/params"
+	"github.com/ethereum/go-ethereum/rpc"
+)
+
+var (
+	sendRawTxConditionalCostMeter       = metrics.NewRegisteredMeter("sequencer/sendRawTransactionConditional/cost", nil)
+	sendRawTxConditionalRequestsCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/requests", nil)
+	sendRawTxConditionalAcceptedCounter = metrics.NewRegisteredCounter("sequencer/sendRawTransactionConditional/accepted", nil)
+)
+
+type sendRawTxCond struct {
+	b      ethapi.Backend
+	seqRPC *rpc.Client
+}
+
+func GetSendRawTxConditionalAPI(b ethapi.Backend, seqRPC *rpc.Client) rpc.API {
+	return rpc.API{
+		Namespace: "eth",
+		Service:   &sendRawTxCond{b, seqRPC},
+	}
+}
+
+func (s *sendRawTxCond) SendRawTransactionConditional(ctx context.Context, txBytes hexutil.Bytes, cond types.TransactionConditional) (common.Hash, error) {
```

Comment:
A reference to the contents of cond is taken in an atomic.Pointer. It's not ideal, but it just happens to work because the receiver gets wrapped in the RPC API, which seems to generate owned values and hand them over wholesale to the method here. I can see that TransactionConditional contains pointers for optional values so that must be why it's done like this.

Thought I should flag it in case it wasn't intentional, but I think it's fine.

Reply:
yeah exactly, not ideal but a pattern in go that the transaction is passed around pretty much everywhere by reference and the optional values are populated prior to creating that transaction, then handed off via the atomic pointer on the transaction itself.

Open to changing this though if you feel there's a better method

Reply:
I think it's out of scope for this PR, I think it's going to be the default through the RPC code.

Reply:
okay sounds good, going to resolve this thread